hicommonwealth/commonwealth
# User-Onboarding: Show default username granted by SSO service

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6750

## Description of Changes
When user onboards with an SSO service, we show the default username granted by that service in the welcome onboard modal

## "How We Fixed It"
N/A

## Test Plan

- Enable `FLAG_USER_ONBOARDING_ENABLED` in `.env`.
- Add `type='create-account'` to `<AuthModal/>` instance in `SubLayoutHeader.tsx` or apply this diff
```diff
diff --git a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
index 4564d26340..c58647b700 100644
--- a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
+++ b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
@@ -59,6 +59,7 @@ export const SublayoutHeader = ({
         open={isFeedbackModalOpen}
       />
       <AuthModal
+        type='create-account'
         onClose={() => setIsAuthModalOpen(false)}
         isOpen={isAuthModalOpen}
       />

```
- Create a new SSO account (google/discord/twitter/apple/github or email) and verify you see the welcome onboard modal showup. The modal will only showup for the first few (5) minutes, since the only requirement of the modal is username which all sso services provide (excluding direct email login, which will be done separately in https://github.com/hicommonwealth/commonwealth/issues/6646)
- Now notice that the welcome modal has the username as default, verify this username matches the username on the SSO service you used to signup.
- Verify you can still change the username

Sidenote: if the username is duplicate then user won't be able to save it.

## Deployment Plan
N/A

## Other Considerations
N/A
